styled/segment.go

```diff
@@ -0,0 +1,104 @@
+package styled
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/elves/elvish/eval/vals"
+)
+
+// A styled Segment is a string that has some style applied to it.
+type Segment struct {
+	Style
+	Text string
+}
+
+func (Segment) Kind() string { return "styled-segment" }
+
+// Returns the representation of this Segment. The string can be used to construct
```

Comment:
Start the docstring with `Repr`, so it reads "Repr returns ...". Same for docstrings below.